### PR TITLE
TOOLSREQ-6431: Add Support for xrefs to Admonitions 

### PR DESCRIPTION
**Pull request by nadamsoreilly**
NOTE: This seems to work in practice but two xspec tests are not working. Creating a draft PR so I don't forget, but this is not ready for review.

Adding support for cross-references to admonitions (notes, tips, warnings) by
1) Giving them the pagenum class in param.xsl so that xrefs in PDFs display
the page number (as dictated by the CSS theme), and 2) adding handling in
xrefgen.xsl so that a warning is display if the linked admonition has no
title. This is necessary because admonitions are valid with and without
titles.

I branched this off of master and it looks like the staging/master commit history diverged during our last rollback. Not sure what to do about that (if anything). As mentioned above, the relevant files for the diff are param.xsl and xrefgen.xsl.

Priority: Low - Kristen mentioned that these references are rare and we have the one off in case it's necessary.

See [TOOLSREQ-6431](https://intranet.oreilly.com/jira/browse/TOOLSREQ-6431)

### CODE REVIEW

**Comment by nadamsoreilly** [comment]
UPDATE: The opfc.xspec and ncx.xpsec tests were failing, but they are also failing for master. That requires some additional investigation, but doesn't seem relevant to this PR.

**Comment by hackermatthew** [comment]
I see your note about tests, but did you write a new one for this change?

**Comment by nadamsoreilly** [comment]
@hackermatthew That's a good point, I had meant to add new tests, but forgot in the interim. I pushed a new commit with new tests added to xrefgen.xspec.

File: htmlbook-xsl/xspec/xrefgen.xspec [hunk 1]
```diff
@@ -1657,5 +1657,114 @@ and the script from https://github.com/xspec/xspec.">
     </x:call>
     <x:expect label="Trimmed URL should be returned as is">oreilly.com</x:expect>
   </x:scenario>
-  
+
+<!-- TOOLSREQ-6431 XREFS for Admonitions begins -->
```
Comment (hackermatthew) [inline htmlbook-xsl/xspec/xrefgen.xspec:1661]: Ticket numbers in comments: are they useful? :) I can't think of a concrete problem with them, I just know I end up deleting them a lot in our other codebases. The ticket reference might allow us to figure out who wanted this change later, but it seems like the tests themselves explain pretty well the intention of the change (or maybe, the tests _should_).

**Review by hackermatthew — APPROVED** [review]

File: htmlbook-xsl/xspec/xrefgen.xspec (hunk 1 above)
Comment (nadamsoreilly) [reply htmlbook-xsl/xspec/xrefgen.xspec]: That's a good point. Considering ticket numbers are not used elsewhere in the XSPEC codebase I went ahead and removed it.

**Review by hackermatthew — APPROVED** [review]
Sorry, I should have approved this last week.

**Pull request merged**